protomap/proto.go

```diff
@@ -920,6 +939,12 @@ func hasValuePathPrefix(opts []UnmapOpt) (*gpb.Path, error) {
 // type of payload is provided for the message. The second, boolean, return argument specifies whether
 // the message provided was a known wrapper type.
 func makeWrapper(msg protoreflect.Message, fd protoreflect.FieldDescriptor, val interface{}) (protoreflect.Message, bool, error) {
+	// If this field was a repeated then it could be a typed value -- but this is handled
+	// separately, thus we simply return false here.
+	if fd.IsList() {
+		return nil, false, nil
+	}
+
```

Comment:
Since makeWrapper is only called in once place, this means that this is triggered when,
1. the proto field is `repeated`.
2. YANG field is not a `leaf-list`.

I believe we must then be dealing with a YANG list. In this case, `makeWrapper` already handles this case, so I think we can remove this if statement.

Reply:
Well spotted, yes, we can -- previously, this was causing a panic because we got to the stage where we called `makeWrapper` without those checks (the leaf-list + leaf-list of unions one). I thought about removing it, but because the consequences are a panic then it seemed like it might be better to be defensive just here. WDYT?

Reply:
Where was it panicking before? I'm ok either way. I tend to let the error happen in cases like these in order to test my understanding of the code, since if the code isn't behaving as I expect then it might be better than a silent error, but I've also ran into situations where that just took unnecessary debugging time later because the defensive code would've been correct.

Reply:
Maybe the most defensive thing to do would be to make it return an error now this should be handled properly. It was throwing the panic at 959 since you can't call `.Message()` on a `repeated` field.

I'll make it return an error -- which we can't easily exercise, but it will avoid the panic whilst surfacing this.

Reply:
LGTM, thanks.